includes/class-newspack-blocks-caching.php

```diff
@@ -0,0 +1,151 @@
+<?php
+/**
+ * Newspack block caching layer
+ *
+ * @package Newspack_Blocks
+ */
+
+/**
+ * Cache dynamic blocks for improved performance.
+ */
+class Newspack_Blocks_Caching {
+	/**
+	 * Add hooks and filters.
+	 */
+	public static function init() {
+		if ( defined( 'NEWSPACK_CACHE_BLOCKS' ) && NEWSPACK_CACHE_BLOCKS ) {
```

Comment:
Nit: the prefix should match the repository name:
```suggestion
		if ( defined( 'NEWSPACK_BLOCKS_CACHE_BLOCKS' ) && NEWSPACK_BLOCKS_CACHE_BLOCKS ) {
```

Reply:
https://github.com/Automattic/newspack-blocks/pull/1772/commits/4b570413299b5c22aabf9973d59290fed8b4fcc0